### PR TITLE
Feature/xr keyboard

### PR DESCRIPTION
**Pull request by andybak**
Including a basic "rename sketch" feature.

image

### CODE REVIEW

File: Assets/Scripts/GUI/KeyboardPopUpWindow.cs [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+
+using System;
+using System.IO;
+
+namespace TiltBrush
+{
+    public class KeyboardPopUpWindow : OptionsPopUpWindow
+    {
+        private KeyboardUI m_KeyboardUI;
+        private const string m_validFilenameCharacters = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789._-";
```
Comment (TimAidley) [inline Assets/Scripts/GUI/KeyboardPopUpWindow.cs:23]: This doesn't appear to be used - is it meant to be?

**Review by TimAidley — APPROVED** [review]
At last! a keyboard in Open Brush! LGTM!

File: Assets/Scripts/GUI/KeyboardPopUpWindow.cs (hunk 1 above)
Comment (andybak) [reply Assets/Scripts/GUI/KeyboardPopUpWindow.cs:23]: Good spot. I'll remove it

Comment (andybak) [reply Assets/Scripts/GUI/KeyboardPopUpWindow.cs]: See 420af0317d5c12f95c4233b71157f6c8ce6af3e9

**Comment by andybak** [comment]
The key name translations are GPT generated. I think they are probably ok as I checked a sample and they seemed good. But would be worth getting them double checked by a native speaker.

File: Assets/Settings/Localization/Strings/Strings_en.asset [hunk 2]
```diff
@@ -609,7 +609,7 @@ MonoBehaviour:
     m_Metadata:
       m_Items: []
   - m_Id: 7870824155160576
-    m_Localized: Snap and Stroke Settings
+    m_Localized: Grid and Snap Settings
```
Comment (mikeage) [inline Assets/Settings/Localization/Strings/Strings_en.asset:612]: Was this change on purpose?

File: Assets/Settings/Localization/Strings/Strings_es.asset [hunk 3]
```diff
@@ -609,7 +609,7 @@ MonoBehaviour:
     m_Metadata:
       m_Items: []
   - m_Id: 7870824155160576
-    m_Localized: Ajustes de cambio y trazo
+    m_Localized:
```
Comment (mikeage) [inline Assets/Settings/Localization/Strings/Strings_es.asset:612]: Is this deliberately removed? (there are a few more like this; I'm not highlighting all of them

**Review by mikeage — COMMENTED** [review]
Please check
